### PR TITLE
Introduce bower runner

### PR DESCRIPTION
**Pull request by vicmosin**
Couldn't really test it as since couldn't install development environment properly..

### CODE REVIEW

File: .idea/runConfigurations/Server.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <option name="COMMON_VM_ARGUMENTS" value="-ea -Xmx384m -XX:MaxPermSize=200m -Ddebug=true -server -XX:+HeapDumpOnOutOfMemoryError -Djava.awt.headless -DTC.res.disableAll=true -Dteamcity.development.mode=true -Dteamcity.development.shadowCopyClasses=true -Dteamcity.data.path=$TeamCityDistribution$/../PluginDev.Data -Dteamcity_logs=$TeamCityDistribution$/logs -Dlog4j.configuration=file:$TeamCityDistribution$/conf/teamcity-server-log4j.xml" />
     <option name="UPDATING_POLICY" value="restart-server" />
     <deployment>
-      <file path="$TeamCityDistribution$/webapps/ROOT">
+      <file path="$APPLICATION_HOME_DIR$/bin/Users/victormosin/Downloads/TeamCity-10.0.4.tar.gz/webapps/ROOT">
```
Comment (jonnyzzz) [inline .idea/runConfigurations/Server.xml]: please revert it and rename JDK setup locally. This is likely to change build

**Comment by vicmosin** [comment]
Could you please help me to build it? I can't do that even if I pointed the project to TeamCity.tar file and gave hime a tomcat7 instance..

**Comment by jonnyzzz** [comment]
It was just discussed in a recent issue. Tomcat support in intellij is only
available in untimate edition. You may download an intellij idea ultimate
eap for that or use evaluation.

Could you please log you trials and yhat improve readme based on your
experience. Looks like I am quite biased for that.

File: .idea/runConfigurations/Server.xml (hunk 1 above)
Comment (vicmosin) [inline .idea/runConfigurations/Server.xml]: @jonnyzzz  I guess I did it already..

**Comment by vicmosin** [comment]
Ok, I will try. Another issue is that it can't determine all jetbrains's namespaces..
for example, `import jetbrains.buildServer.agent.AgentBuildRunnerInfo`

**Comment by jonnyzzz** [comment]
That is normal. You should setup TEAMCITY_DISTRIBUTION_PATH path variable
in idea settings to point to unpacked tar.gz TeamCity distribution.

As far as I remember this part should be in the Readme.

Also it's ok to move project to Gradle. There is gradle plugin that helps
pack plugin for TeamCity.

**Comment by vicmosin** [comment]
@jonnyzzz I could finally compile it. Had to add all the dependencies manually, link to teamcity's tar file did not help. Anyway, now I try to run the Server locally but it fails with:
```
Error:Internal error: (java.io.FileNotFoundException) /Users/Downloads/TeamCity-10.0.4.tar.gz/../PluginDev.Data/plugins/jonnyzzz.node.zip (Not a directory)
java.io.FileNotFoundException: ...
```
shouldn't it deploy zip file with plugin automatically?

**Comment by vicmosin** [comment]
And another question.. I found that zip file was generated at out directory of project. In order to test it on real machine, what exactly should I do? I have that plugin installed, how can I replace it without removing all existing runners?

**Comment by jonnyzzz** [comment]
You may  call `build | articats` and select there to build a plugin `.zip` file. Next you may place this .zip file into TeamCity server data directory and restart it. 

There used to be a run configuration that automates the process. But it was quite fragile setup and it may be broken on your setup. Anyway it was only supported for IntelliJ Ultimate

To automate deployment you may have either a symlink or a server start script that would grab created .zip file and place it onto your `.buildserver` directory

**Comment by vicmosin** [comment]
@jonnyzzz 
ok, I finally tested the updated version of plugin (with my changes) on a real instance, looks fine.

**Comment by jonnyzzz** [comment]
👍

**Pull request merged**